### PR TITLE
Add l402 support

### PR DESCRIPTION
**Pull request by SamuelAl**
This PR adds L402 support to all OrdinalsBot modules, based on Sulu's L402 ordinalsbot host: ordinalsbot.ln.sulu.sh.

- The L402 support does not affect users that want to use the API using a traditional API key
- Support is added by using the `l402` [npm package ](https://www.npmjs.com/package/l402) to wrap the axios instances with an L402 interceptor. This interceptor handles L402 errors and payments transparently.
- Example using L402 included in examples folder
- All L402 functionality is unit tested
- The l402 package includes a wallet implementation using Alby. Users are free to create their own Wallet implementations depending on the lightning wallets they want to use (we can support this as well). They just have to conform to the Wallet interface.

### CODE REVIEW

**Comment by ordinariusprof** [comment]
@SamuelAl can you send this PR to dev branch? and bump the package version as well please?
please also check for conflicts.

File: src/launchpad/client.ts [hunk 1]
```diff
@@ -63,18 +69,23 @@ export class LaunchpadClient {
...
       }
 
+      // Choose the base URL based on whether L402 is used or not
+      const baseURL = options?.useL402
+        ? "https://ordinalsbot.ln.sulu.sh/runes/launchpad/"
+        : this.env === "live"
+          ? "https://api.ordinalsbot.com/runes/launchpad/"
+          : "https://testnet-api.ordinalsbot.com/runes/launchpad/";
+
+      // Create the Axios client with the appropriate base URL
```
Comment (T1-O) [inline src/launchpad/client.ts:79]: This is not runes/launchpad, it should be:
          this.env === "live"
            ? `https://api.ordinalsbot.com/launchpad/`
            : `https://testnet-api.ordinalsbot.com/launchpad/`,

File: examples/l402.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import { AlbyWallet, MemoryTokenStore } from "l402";
+const { Client } = require('@getalby/sdk');
+import { ClientOptions, Satscanner } from "../src";
+require('dotenv').config();
+
+// Load the Alby token from an environment variable
```
Comment (T1-O) [inline examples/l402.ts:6]: The example is good.
Would be great to have a section in the main readme for L402 and link to this example.

Comment (T1-O) [reply examples/l402.ts:6]: This could also have links or a summary explaining L402 in general

File: src/client.ts [hunk 3]
```diff
@@ -40,35 +41,46 @@ export class InscriptionClient {
...
-          environment === "live"
-            ? `https://api.ordinalsbot.com`
-            : `https://testnet-api.ordinalsbot.com`,
-        timeout: 30000,
-        headers: {
-          "x-api-key": this.api_key,
-          Connection: "Keep-Alive",
-          "Content-Type": "application/json",
-          "Keep-Alive": "timeout=10",
-          "User-Agent": packageVersion,
```
Comment (T1-O) [inline src/client.ts:62]: Package version is missing from updated headers

File: examples/l402.ts (hunk 2 above)
Comment (SamuelAl) [reply examples/l402.ts:6]: That is a good idea.

Comment (SamuelAl) [reply examples/l402.ts:6]: Added that to the README.

File: src/client.ts (hunk 3 above)
Comment (SamuelAl) [reply src/client.ts:62]: Done.

**Comment by T1-O** [comment]
Thanks, @SamuelAl This would be a nice addition.
1. Do we need to manage any configuration/account etc for using this URL? https://ordinalsbot.ln.sulu.sh/
2. We will merge this to dev now, we can do some testing to ensure the SDK changes are backwards compatible
3. Would be great if you can do some testing with the L402 option.

**Review by T1-O — APPROVED** [review]
Thanks for the PR, this will be a good enhancement 👍

**Comment by SamuelAl** [comment]
Great @T1-O !

We are currently managing the ordinalsbot account in our platform. At some point we should pass it over to you guys so you can have complete control of things like pricing etc, but we are happy to keep managing it during the trial phase. 

Please do tell us if any issues arise during testing, we will do L402 testing ourselves.

Thanks!

**Pull request merged**